main.go

```diff
@@ -191,9 +196,20 @@ func registerMetricsRouter(router *mux.Router) error {
 	return nil
 }
 
+// getHealthCheckURL - extracts the health check URL.
+func getHealthCheckURL(endpoint, healthCheckPath string, healthCheckPort int) string {
+	healthCheckURL, _ := url.Parse(strings.TrimSuffix(endpoint, slashSeparator) + healthCheckPath)
+	if healthCheckPort != 0 {
```

Comment:
This function can return early when healthCheckPort is not set without resorting to url.Parse, passing port is rare so we should keep the code as such.

Reply:
Updated the code using a fast path when the `healthCheckPort` is set to `0`